x/merkledb/codec.go

```diff
@@ -81,7 +84,7 @@ type codecImpl struct {
 }
 
 func (c *codecImpl) encodeDBNode(n *dbNode) []byte {
-	buf := &bytes.Buffer{}
+	buf := bytes.NewBuffer(make([]byte, 0, defaultNodeAllocSize))
```

Comment:
I actually wonder if it's possible to get a tighter estimate on this preallocation. the `dbNode` has two fields which is a `maybe.Maybe[[]byte]` and a `map[byte]child` where `child` is type of `path (string)` and `ids.ID (32 bytes)`.

Is it possible for us to do add up the lengths of these at runtime and that would give us a better allocation estimate..? I'm not super familiar with this code so I ignore me if I'm saying something that sounds strange X)